zkevm-circuits/src/copy_circuit.rs

```diff
@@ -689,14 +697,22 @@ pub mod dev {
 mod tests {
     use super::dev::test_copy_circuit;
     use bus_mapping::evm::{gen_sha3_code, MemoryKind};
-    use bus_mapping::{circuit_input_builder::CircuitInputBuilder, mock::BlockData};
+    use bus_mapping::{
+        circuit_input_builder::{CircuitInputBuilder, CircuitsParams},
+        mock::BlockData,
+    };
     use eth_types::{bytecode, geth_types::GethData, Word};
     use mock::test_ctx::helpers::account_0_code_account_1_no_code;
     use mock::TestContext;
 
     use crate::evm_circuit::test::rand_bytes;
     use crate::evm_circuit::witness::block_convert;
 
+    const CIRCUITS_PARAMS: CircuitsParams = CircuitsParams {
```

Comment:
We are use `CircuitsParams::default()` and constants CIRCUITS_PARAMS everywhere. What do you think we define some constants like CIRCUITS_PARAMS_S(256 rws), CIRCUITS_PARAMS_M(512 rws), CIRCUITS_PARAMS_L(2048 rws), CIRCUITS_PARAMS_XL(8192 rws) in a file? The benefit would be 

1. less ad hoc magic number scattering around.
2. Easy to search and track what testing cases are using large params.

Reply:
Note: In https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/795/commits/bc2d34c9b51d9254c7e1f5c70c6b4cd87e0d04e6 I've cleaned up some of these `CIRCUITS_PARAMS` because one was not used, and the other once was only used once and didn't need to be `const`.

About your suggestion: Initially I thought about doing something similar to what you mention, but then I found that there are many configurations of particular cases that would only be used once or twice, so I just defined a default that works for 90% of the cases, and for the rest I thought declaring the parameters in the test itself is not that bad.  For example, notice that one test requires `max_txs = 4`, so we would need 5 different constants for parameters sets.

It's true that this way it would be easy to track tests using large params, but I think currently we could just find those that are not using the default params and they would be the large ones.